src/IW/Sync/Search.hs

```diff
@@ -0,0 +1,69 @@
+module IW.Sync.Search
+       ( fetchAllHaskellRepos
+       , fetchAllHaskellIssues
+       , fetchOpenIssuesByLabels
+       ) where
+
+import GitHub (SearchResult (..), Error (..))        
+import GitHub.Endpoints.Search (searchRepos, searchIssues)
+
+import IW.App (WithError)
+import IW.Core.Issue (Issue (..), Label (..))
+import IW.Core.Repo (Repo (..), RepoOwner (..), RepoName (..))
+import IW.Core.SqlArray (SqlArray (..))
+import IW.Db (WithDb)
+import IW.App.Error (AppErrorType, throwError)
+
+import qualified GitHub
+import qualified Data.Vector as Vector
+
+
+-- | Fetch all repositories with Haskell language
+fetchAllHaskellRepos :: (MonadIO m, WithDb env m, WithError m) => m [Repo]
+fetchAllHaskellRepos = do
+    result <- liftIO $ searchRepos "language:haskell"
```

Comment:
Code is a bit ad-hoc in this module. Let's make it cleaner. I propose to move some common and relevant parts into separate functions, so types can help us.

First, let's create function:

```haskell
searchHaskellRepos :: IO ...
searchHaskellRepos = searchRepos ...
```

I think it's useful to see the explicit type of this function and what it does.

Then, our search GitHub functions always will return some `Either ... SearchResult`. Let's introduce helper function:

```haskell
liftGitHubSearchToApp :: ... m => IO (Either ... (SearchResult a)) -> m [a]
```

This function will convert `Vector` to list and `throwError` when we see an error. This function also can do some useful logging since `SearchResult` contains total count, so we can log something like:

```
Fetching total of 100500 Repos...
```